state/indexer/block/kv/kv.go

```diff
@@ -34,13 +36,44 @@ type BlockerIndexer struct {
 	// Add unique event identifier to use when querying
 	// Matching will be done both on height AND eventSeq
 	eventSeq int64
-	log      log.Logger
+	// Lower bound for the last retained height
+	lbRetainedHeight int64
+	log              log.Logger
 }
 
 func New(store dbm.DB) *BlockerIndexer {
 	return &BlockerIndexer{
-		store: store,
+		store:            store,
+		lbRetainedHeight: getLBRetainHeight(store),
+	}
+}
+
+func getEventsForHeightKey(height int64) []byte {
+	return []byte(fmt.Sprintf("eventsForHeightKey%d", height))
```

Comment:
As mentioned below, you don't need a new key for this, you can reuse the existing height key , just instead of storing the height as the value (which is done right now), you store the new data.